### ISSUE TITLE
[__get_names] Returns wind speed and vertical wspd cols

### ISSUE THREAD

**Issue by rach185** [brightwind-dev/brightwind#450]
**Describe the bug**
When using the private function `__get_names`, if the `measurement_type_id = 'vertical_wind_speed'`, then all wind speed and vertical wind speed columns names are returned. 

**To Reproduce**

```
meas_sta_uuid = 'f4af1419-147e-420d-b71a-431df3bf4fe2'
meas_station = bw.MeasurementStation(bw.LoadBrightHub.get_data_model(measurement_station_uuid=meas_sta_uuid))
measurement_type_id= 'vertical_wind_speed'
meas_station.measurements._Measurements__get_names(measurement_type_id)
```

**Expected behavior**
The wind speed columns should not be returned when the `measurement_type_id` is `vertical_wind_speed`. If the  `measurement_type_id = ['vertical_wind_speed']` then the operation is as expected.

**Screenshots**
![image](https://github.com/user-attachments/assets/bf641a66-a726-4737-b871-759fc3224371)

**Comment by BiancaMorandi** [comment]
I think that this function shouldn't be private as it is useful to be able to extract the measurement names by measurement type

**Comment by olivia-bentley** [comment]
work on this

**Comment by stephenholleran** [comment]
> [@stephenholleran](https://github.com/stephenholleran) I think that this function shouldn't be private as it is useful to be able to extract the measurement names by measurement type, what do you think?

It could be. So it would be used as outlined below?
`meas_station.measurements.get_names(measurement_type_id)`

**Comment by BiancaMorandi** [comment]
> It could be. So it would be used as outlined below?
> `meas_station.measurements.get_names(measurement_type_id)`

yes, it can be used as for your suggestion, or another option is to update the `meas_station.measurements.names` function to accept measurement type as an option. The `meas_station.measurements.names` function at the moment is using the private `_get_names` function for pulling out all meas names but the measurement_type is to NOT given as an optional input. I suggest the below, @stephenholleran are you happy with this?
- update `meas_station.measurements.names` to be `meas_station.measurements.get_names`
- add option of giving measurement_type as input, if measurement_type=None then pull out all meas names as for current behaviour of `meas_station.measurements.names`

**Comment by stephenholleran** [comment]
Ah sorry, I didn't think of that. Just update the `names()` to accept the measurement type. That should be easier.

**Comment by BiancaMorandi** [comment]
AS FOR DISCUSSION NOT DEPREC WARDING ENEDEED FOR NEAMES, ONLLY POINT 2 OF MY COMMENT NEEDED

**Comment by BiancaMorandi** [comment]
@stephenholleran merge request for this issue is ready for your review/approval. As part of this issue we have addressed the below:
- fixed the bug
- changed `_Measurements_get_names` to be public
- updated docstring with example usage and added tests for `get_names` and 'get_heights' functions
- on `test_station.py` I have also moved the importing of demo json datasets outside test functions because these are used by several test functions so we load these for tests only once
- updated the changelog, note that I have slightly reworded the description for latest issue we have merged to `dev` (for iss421) so need to make sure that latest wording is kept on `dev` when merging
